evm_test_runner/src/plonky2_runner.rs

```diff
@@ -296,25 +296,29 @@ fn run_test_or_fail_on_timeout(
 
 /// Run a test against `plonky2` and output a result based on what happens.
 fn run_test_and_get_test_result(test: TestVariantRunInfo) -> TestStatus {
-    if TryInto::<u32>::try_into(test.gen_inputs.block_metadata.block_gaslimit).is_err() {
-        // Gas limit of more than 32 bits is not supported by the zkEVM.
-        return TestStatus::Ignored;
-    }
-
     let timing = TimingTree::new("prove", log::Level::Debug);
 
     let proof_run_res = prove_with_outputs::<GoldilocksField, KeccakGoldilocksConfig, 2>(
         &AllStark::default(),
         &StarkConfig::standard_fast_config(),
-        test.gen_inputs,
+        test.gen_inputs.clone(),
         &mut TimingTree::default(),
     );
 
     timing.filter(Duration::from_millis(100)).print();
 
     let (proof_run_output, generation_outputs) = match proof_run_res {
         Ok(v) => v,
-        Err(evm_err) => return TestStatus::EvmErr(evm_err.to_string()),
+        Err(evm_err) => {
+            if format!("{:?}", evm_err).contains("GasLimitError")
+                && TryInto::<u32>::try_into(test.gen_inputs.block_metadata.block_gaslimit).is_err()
+            {
+                // Gas limit of more than 32 bits is not supported by the zkEVM.
+                return TestStatus::Ignored;
+            } else {
+                return TestStatus::EvmErr(evm_err.to_string());
+            };
+        }
```

Comment:
Looks good! One nit though. I think it's a bit cleaner to use `to_string` instead of a format with the debug format `{:?}` (Like this: `if evm_err.to_string().contains("GasLimitError")`).

Reply:
You're right, I just updated it!